applications/StructuralMechanicsApplication/custom_advanced_constitutive/generic_small_strain_high_cycle_fatigue_law.cpp

```diff
@@ -77,60 +78,87 @@ void GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::CalculateMa
         double uniaxial_stress;
         TConstLawIntegratorType::YieldSurfaceType::CalculateEquivalentStress(predictive_stress_vector, r_strain_vector, uniaxial_stress, rValues);
 
-        double max_stress = this->GetMaxStress();
-        double min_stress = this->GetMinStress();
-        double sign_factor = HighCycleFatigueLawIntegrator<6>::CalculateTensionCompressionFactor(predictive_stress_vector);
-        uniaxial_stress *= sign_factor;
-        unsigned int global_number_of_cycles = this->GetNumberOfCyclesGlobal();
-        unsigned int local_number_of_cycles = this->GetNumberOfCyclesLocal();
-        bool max_indicator = this->GetMaxDetected();
-        bool min_indicator = this->GetMinDetected();
-        double fatigue_reduction_factor = this->GetFatigueReductionFactor();
-        double B0 = this->GetFatigueReductionParameter();
-        double previous_max_stress = this->GetPreviousMaxStress();
-        double previous_min_stress = this->GetPreviousMinStress();
-        double wohler_stress = this->GetWohlerStress();
-
+        double max_stress = mMaxStress;
+        double min_stress = mMinStress;
+        unsigned int global_number_of_cycles = mNumberOfCyclesGlobal;
+        unsigned int local_number_of_cycles = mNumberOfCyclesLocal;
+        bool max_indicator = mMaxDetected;
+        bool min_indicator = mMinDetected;
+        double fatigue_reduction_factor = mFatigueReductionFactor;
+        double B0 = mFatigueReductionParameter;
+        double previous_max_stress = mPreviousMaxStress;
+        double previous_min_stress = mPreviousMinStress;
+        double wohler_stress = mWohlerStress;
+        double reversion_factor_relative_error = mReversionFactorRelativeError;
+        double max_stress_relative_error = mMaxStressRelativeError;
+        double CyclesToFailure = mCyclesToFailure;
+        bool new_cycle = false;
+        double s_th = mThresholdStress;
+        unsigned int cycles_after_advance_strategy = mCyclesAfterAdvanceStrategy;
+        bool adnvance_strategy_applied = rValues.GetProcessInfo()[ADVANCE_STRATEGY_APPLIED];
+        bool damage_activation = rValues.GetProcessInfo()[DAMAGE_ACTIVATION];
```

Comment:
In the future could be nice to create an structure called HCF_parameters or something like that in order to better encapsulate all this features